class-02-jquery-and-events/pair-assignment/starter-code/scripts/articleView.js

```diff
@@ -0,0 +1,109 @@
+// Configure a view object, to hold all our functions for dynamic updates and article-related event handlers.
+var articleView = {};
+
+articleView.populateFilters = function() {
+  $('article').each(function() {
+    if (!$(this).hasClass('template')) {
+      // DONE: We need to take every author name from the page, and make it an option in the Author filter.
+      //       To do so, Build an `option` DOM element that we can append to the author select box.
+      //       Start by grabbing the author's name from `this` article element, and then use that bit of
+      //       text to create the option tag (in a variable named `optionTag`),
+      //       that we can append to the #author-filter select element.
+      //       YAY, DOM manipulation!
+      var val = $(this).find('address a').text();
+      var optionTag = '<option value="' + val + '">' + val + '</option>';
+      $('#author-filter').append(optionTag);
+
+      // DONE: Similar to the above, but...
+      //       Avoid duplicates! We don't want to append the category name if the select
+      //       already has this category as an option!
+      val = $(this).attr('data-category');
+      optionTag = '<option value="' + val + '">' + val + '</option>';
+      if ($('#category-filter option[value="' + val + '"]').length === 0) {
+        $('#category-filter').append(optionTag);
+      }
+    }
+  });
+};
+
+articleView.handleAuthorFilter = function() {
+  $('#author-filter').on('change', function() {
+    if ($(this).val()) {
+
+      // DONE: If the select box was changed to an option that has a value, we need to hide all the articles,
+
+      $('article').hide();
+
+      /* and then show just the ones that match for the author that was selected. Use an "attribute selector" to find those articles, and fade them in for the reader.*/
+
+      $('article[data-author="' + $(this).val() + '"]').fadeIn();
+
+    } else {
+      $('article').show();
+      $('.template').hide();
+      // DONE: If the select box was changed to an option that is blank, we should
+      //       show all the articles, except the one article we are using as a template.
+
+    }
+    $('#category-filter').val('');
+  });
+};
+
+articleView.handleCategoryFilter = function() {
+  // DONE: Just like we do for #author-filter above, we should handle change events on the #category-filter element.
+  //       When an option with a value is selected, hide all the articles, then reveal the matches.
+  //       When the blank (default) option is selected, show all the articles, except for the template.
+  //       Be sure to reset the #author-filter while you are at it!
+  $('#category-filter').on('change', function() {
+    if ($(this).val()) {
+      $('article').hide();
+      $('article[data-category="' + $(this).val() + '"]').fadeIn();
+    } else {
+      $('article').show();
+      $('.template').hide();
+    }
+    $('#category-filter').val('');
```

Comment:
This line should be setting the author-filter value to a blank string rather than the category filter.  This is why you are not seeing the category you have selected in the option pane after you have selected a category.  And instead you are still seeing an author name selected in the author filter pane.